### PR TITLE
fix](decimal) fix Arithmetic Overflow error of converting string to decimal

### PR DESCRIPTION
**Pull request by jacktengg**
## Proposed changes

Issue Number: pick from #32246

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

**Comment by jacktengg** [comment]
run buildall

**Comment by jacktengg** [comment]
run buildall

File: regression-test/suites/datatype_p0/decimalv3/test_load.groovy [hunk 1]
```diff
@@ -104,4 +104,57 @@ suite("test_load") {
...
+    """
+    sql """
+        insert into test_sys_update_basic_test_update_decimal_tb values
+            (1.001, 2.002), (1.002, 0.00000002), (1.003, 0.100000001), (1.004, 0.100044001), (1.005, 0.100045001);
+    """
+    qt_decimalv3_underflow3 """
+        select * from test_sys_update_basic_test_update_decimal_tb order by 1, 2;
+    """
+    sql """
+        UPDATE test_sys_update_basic_test_update_decimal_tb SET v1="0.00000001" WHERE k1 = 1.001;
```
Comment (cambyzju) [inline regression-test/suites/datatype_p0/decimalv3/test_load.groovy:155]: could you use a different literal to test update?

I patched this pr, and try to reproduce again.
Now update stmt execute success, but the result of v1 do not changed, still is "2.00200"

Comment (jacktengg) [reply regression-test/suites/datatype_p0/decimalv3/test_load.groovy]: Seems it's problem of old planner, I updated the test case.

**Comment by jacktengg** [comment]
run buildall

File: regression-test/suites/datatype_p0/decimalv3/test_load.groovy (hunk 1 above)
Comment (cambyzju) [reply regression-test/suites/datatype_p0/decimalv3/test_load.groovy]: > Seems it's problem of old planner, I updated the test case.

Master with old planner;
2.0 both old planner and nereids;

File: regression-test/suites/datatype_p0/decimalv3/test_load.groovy [hunk 2]
```diff
@@ -104,4 +104,59 @@ suite("test_load") {
...
+        CREATE TABLE test_sys_update_basic_test_update_decimal_tb (
+          k1 DECIMAL(10, 5) NULL, 
+          v1 DECIMAL(10, 5) NULL
+        ) UNIQUE KEY(k1) DISTRIBUTED BY HASH(k1) BUCKETS 5 PROPERTIES (
+          "replication_allocation" = "tag.location.default: 1"
+        );
+    """
+    sql """
+        insert into test_sys_update_basic_test_update_decimal_tb values
+            (1.001, 2.002), (1.002, 0.00000002), (1.003, 0.100000001), (1.004, 0.100044001), (1.005, 0.100045001);
```
Comment (cambyzju) [inline regression-test/suites/datatype_p0/decimalv3/test_load.groovy:123]: Insert:
0.00000002 ===> 0.00000

Update:
"0.00000001" ===> 0.00000

still the same result, we still do not know `update set` stmt work correct or not.

maybe insert 1.00000001, then update 2.00000002 ?

**Review by cambyzju — APPROVED** [review]
LGTM

**Pull request merged**